### PR TITLE
Tools: put Mission files into right directories

### PR DESCRIPTION
**Pull request by khancyr**
NFC: this put the missions files for test into their respectives directory to cleanup a little the autotest directory.

### CODE REVIEW

**Comment by peterbarker** [comment]
Thanks, I was considering doing this earlier today, in fact!

Couple of things... if we do go down the path you've gone down here:
 - `load_mission` should just take the name of the mission
 - it should be factored to be a thin wrapper around a `load_mission_from_filepath`

However, I was wondering if we would actually be better off doing per-test subdirectories rather than per-mission-item-type directories.  So we'd have something like 

`Tools/autotest/copter/LoiterToAlt/copter_loiter_to_alt.txt` (or, perhaps just `Tools/autotest/copter/LoiterToAlt/mission.txt`

Separate directories for missions, fences and rally points?

My reasoning is that apart from all being (e.g.) missions, there's nothing to relate the mission files.  For a particular test, there may be value in grouping together all of the files used by the test.

**Comment by khancyr** [comment]
I will do the change for load_mission.

About putting into test named directories .. why not. But that could be complex do to the fences and rally that are shared.
I will try to change a little, we can adjust later, my goal was just to stop having a lot of file on the autotest directory :-P

**Comment by khancyr** [comment]
@peterbarker I have done it for rover. What do you think ?
Load_mission will now directly search the file in the test named directory. Maybe we can rename all test files as mission.txt for easier loading, that way, we just have to do load_mission without any parameters as it will be the same everywhere!

File: Tools/autotest/apmrover2.py [hunk 1]
```diff
@@ -4633,8 +4635,8 @@ def test_poly_fence_object_avoidance_auto_bendy_ruler_easier(self, target_system
         if not self.mavproxy_can_do_mision_item_protocols():
             return
 
-        self.load_fence("rover-path-bendyruler-fence.txt")
-        self.load_mission("rover-path-bendyruler-mission-easier.txt")
+        self.load_fence("rover_path_bendyruler_fence_easier.txt")
```
Comment (peterbarker) [inline Tools/autotest/apmrover2.py]: Please don't canonicalise as part of this PR :-)

**Comment by peterbarker** [comment]
I've rebased this and force-pushed it up to see if it passes CI.  I'm happy to merge it if it passes CI - but Sam's changed might have killed this.

**Comment by khancyr** [comment]
@peterbarker it pass CI and I have put another commit to correct soaring test !

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**